src/func_surf_energy_bal.c

```diff
@@ -688,6 +691,32 @@ double func_surf_energy_bal(double Ts, va_list ap)
   else
     Ra_used[0] = HUGE_RESIST;
   
+  if (veg_var->vegcover < 1) {
+    if (Ra_used[0] > 0) {
+      ga_veg = 1/Ra_used[0];
+      tmp_wind[0] = wind[0];
+      tmp_wind[1] = -999.;
+      tmp_wind[2] = -999.;
+      tmp_height = soil_con->rough/0.123;
+      tmp_displacement[0] = calc_veg_displacement(tmp_height);
+      tmp_roughness[0] = soil_con->rough;
+      tmp_ref_height[0] = 10;
+      Error = CalcAerodynamic(0,0,0,soil_con->snow_rough,soil_con->rough,0,Ra_bare,tmp_wind,tmp_displacement,tmp_ref_height,tmp_roughness);
+      Ra_bare[0] /= StabilityCorrection(tmp_ref_height[0], tmp_displacement[0], TMean, Tair, tmp_wind[0], tmp_roughness[0]);
+      if (Ra_bare[0] > 0) {
+        ga_bare = 1/Ra_bare[0];
+        ga_average = veg_var->vegcover*ga_veg + (1-veg_var->vegcover)*ga_bare;
+        Ra_used[0] = 1/ga_average;
+      }
+      else {
+        Ra_used[0] = 0;
+      }
+    }
+    else {
+      Ra_used[0] = 0;
+    }
+  }
+
```

Comment:
Still needs those comments. The hardcoded values need to be moved to header files.